script/slurm/slurm_nss_matrix.sh

```
@@ -104,15 +104,18 @@ apt-get update && apt-get install -y --no-install-recommends \
 # NSS_PYTHON_VERSION is resolved in env_variables.sh (from .python-version, with
 # an optional override), which is always sourced before this script runs.
 
 if [[ -n "${NSS_VERSION:-}" ]]; then
     # Install nemo-safe-synthesizer from PyPI into a versioned venv cached on
     # lustre so concurrent array jobs can share it without redundant downloads.
+    if [[ ! -f "${LUSTRE_DIR}/.uv/bin/env" ]]; then
+        echo "[NSS SLURM] ERROR: PyPI mode requires uv at ${LUSTRE_DIR}/.uv/bin" >&2
+        echo "[NSS SLURM] Install uv as described in script/slurm/README.md, or omit --nss-version to use repo mode." >&2
+        exit 1
+    fi
+    source "${LUSTRE_DIR}/.uv/bin/env"
```

Shellcheck (0.11.0)

[info] Not following: ./.uv/bin/env was not specified as input (see shellcheck -x).

(SC1091)